crates/sim2h/src/cache.rs

```diff
@@ -208,10 +208,10 @@ impl AspectList {
     }
 
     pub fn add(&mut self, entry_address: EntryHash, aspect_address: AspectHash) {
-        self.0
-            .entry(entry_address)
-            .or_insert_with(Vec::new)
-            .push(aspect_address);
+        let list = self.0.entry(entry_address).or_insert_with(Vec::new);
+        if !list.contains(&aspect_address) {
```

Comment:
this is going to get slower as the list grows, do we want to convert to HashSet? - cool if that's an optimization for later

Reply:
Agreed- just make it a hashset- way more efficient.